app/foodzone/venues/list_retriever.rb

```diff
@@ -0,0 +1,66 @@
+module Venues
+  class ListRetriever
+    def self.execute(venue_repository:)
+      new(venue_repository).execute
+    end
+
+    def initialize(venue_repository)
+      @venue_repository = venue_repository
+    end
+
+    def execute
+      List.new(venue_repository.all)
+    end
+
+    class List
+      EMPTY_MSG = "There are no venues covering that address"
+
+      def initialize(venues)
+        @venues = venues
+      end
+
+      def message
+        EMPTY_MSG if venues.empty?
+      end
+
+      def to_a
+        @rows ||= venues.map do |venue|
+          venue.branches.inject({}) do |hash, branch|
```

Comment:
Unused block argument - `hash`. If it's necessary, use `_` or `_hash` as an argument name to indicate that it won't be used.